### PR TITLE
T/15 Image upload progress UI

### LINKED ISSUE #15: Upload progress UI
There should be a nice UI for the reading and uploading images in the CKEditor 5. It should handle 2 cases:

## Case 1: I'm uploading a small image (1MB with SSD drive)

I this case the user will see the placeholder for the very short time (less than half of the second). The image will load quickly and upload start sooner than user notice that there was a reading step. In this case, we want to avoid showing a big and dark placeholder, what will look like a strange blinking, breaking the feeling of the smooth image uploading.

## Case 2: I'm uploading a big image (5 MB with HDD drive)

I this case the reading process take some time (more-less 3 seconds). We need to show the user something for these 3 seconds because if there will be no reaction after he pasted, dropped or selected the image he will have a feeling that something is broken. We need to show him some kind of a placeholder.

### PR DESCRIPTION
**Pull request by szymonkups**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Feature: Introduced ImageUploadProgress plugin. Closes #15.

### CODE REVIEW

File: src/imageupload.js [hunk 1]
```diff
@@ -23,7 +24,7 @@ export default class ImageUpload extends Plugin {
 	 * @inheritDoc
 	 */
 	static get requires() {
-		return [ ImageUploadEngine ];
+		return [ ImageUploadEngine, ImageUploadProgress ];
```
Comment (pjasiun) [inline src/imageupload.js]: This way we disallow developers to have an upload button but different upload progress. We could move the button component to the separate plugin and make the main upload plugin only aggregates plugins, so do nothing but requires.

Comment (pjasiun) [inline src/imageupload.js]: And of course both `ImageUploadButton` and `ImageUploadProgress` should require `ImageUploadEngine`.

File: src/imageuploadengine.js [hunk 2]
```diff
@@ -85,19 +72,6 @@ export default class ImageUploadEngine extends Plugin {
...
-				return;
-			}
-
-			const modelImage = data.item;
-			const viewFigure = editor.editing.mapper.toViewElement( modelImage );
-			const viewImg = viewFigure.getChild( 0 );
-
-			viewImg.setAttribute( 'src', this.placeholder );
-		} );
 	}
```
Comment (pjasiun) [inline src/imageuploadengine.js]: I might overengineer it, but, isn't it that true that without this converter upload will cause a crash? I believe that engine plugin should work fine, without any additional UI plugin. Then UI plugins should be able to overwrite this very basic converter. Note that the same way there is a converter in the basic styles plugin: https://github.com/ckeditor/ckeditor5-basic-styles/blob/bdc76445715866c948544dc19ff4ad2b569cc302/src/boldengine.js.

**Comment by pjasiun** [comment]
Other than that, it looks beautiful :)

File: src/imageuploadengine.js (hunk 2 above)
Comment (szymonkups) [inline src/imageuploadengine.js]: Engine part will work stand-alone without this converter. Now all upload UI updates can be done using `uploadStatus` attribute as in `ImageUploadProgress`.

**Pull request merged**